microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/HelidonProxyServices.java

```diff
@@ -110,39 +133,75 @@ public boolean supportsClassDefining() {
 
         LOGGER.finest("Defining class " + className + " original class: " + originalClass.getName());
 
+        MethodHandles.Lookup lookup;
+
         try {
-            Module classModule = originalClass.getModule();
-            if (!myModule.canRead(classModule)) {
+            // lookup class name "guessed" from the class name of the proxy
+            String lookupClassName;
+
+            if (className.contains("$")) {
+                // package + first name in the compound proxy class name
+                lookupClassName = className.substring(0, className.indexOf('$'));
+            } else {
+                lookupClassName = className;
+            }
+
+            // I would like to create a private lookup in the same package as the proxied class, so let's do it
+            // first if the producer class and the lookup class name is the same, just use the existing class
+            Class<?> lookupClass = lookupClassName.equals(originalClass.getName()) ? originalClass : null;
+
+            ClassLoader cl = originalClass.getClassLoader();
+
+            if (null == lookupClass) {
+                // try to load the full class name
+                lookupClass = tryLoading(cl, lookupClassName);
+            }
+
+            if (null == lookupClass) {
+                // and if that fails, just use the bean producer class
+                lookupClass = originalClass;
+            }
```

Comment:
Similarly, I think this can all be simplified based on whether `index < 0`.  So:
```
final Class<?> lookupClass = index < 0 ? originalClass : tryLoading(cl, lookupClassName);
```

Reply:
Changed.